src/flags/display.rs

```diff
@@ -45,37 +30,26 @@ impl Configurable<Self> for Display {
         } else if matches.is_present("almost-all") {
             Some(Self::AlmostAll)
         } else if matches.is_present("directory-only") {
-            Some(Self::DirectoryItself)
+            Some(Self::DirectoryOnly)
         } else {
             None
         }
     }
 
     /// Get a potential `Display` variant from a [Config].
     ///
-    /// If the Config's [Yaml] contains a [String](Yaml::String) value pointed to by "display" and
-    /// it is either "all", "almost-all" or "directory-only", this returns the corresponding
-    /// `Display` variant in a [Some]. Otherwise this returns [None].
+    /// If the `Config::display` has value and is one of "all", "almost-all" or "directory-only",
```

Comment:
`VisibleOnly` is much more like a `Default` value in my opinion, and can not be set by arg flags, this is why I did not add it to the comment.

but the missing value in doc-comment also seems like a problem, added as your comment.

As the enum comment part, IMHO, I think we could leave the doc comment as it is now:
1. the `serde.rs` simply deserializes the yaml into Config, although it checked the enum value, it is something like `lexical analysis`
2. `from_config` cares more like `syntactic analysis`, some other logic will  be added here(although it is not for `display`)

but I am open to accepting your idea if you insist.

Reply:
We will have to have a go at redesigning the logic in some places anyway.ike you stated, VisibleOnly can not be set from the CLI args, but this could actually be a problem if it is set to something different in the config, but the user wants to have it VisibleOnly for a single call. Right now there is no way to achieve that aside from modifying the config or ignoring it. But that is outside of the scope of this pull request.

Reply:
https://github.com/Peltoche/lsd/issues/451
issue created